src/dataDisplay/table.jsx

```diff
@@ -1,21 +1,350 @@
+    stackable: undefined,
+    stickyColumnCount: 0,
+    resizableColumnWidthPercentage: undefined,
+    stretch: undefined,
+    striped: undefined,
+    style: undefined,
+};
+
+const DEBOUNCE_WAIT_TIME = 50;
+const TABLE_HEADER_TYPE = 'TableHeader';
+const STICKY_CELL_CLASS = 'sticky-cell';
+const STICKY_CELL_RESIZABLE_CLASS = 'sticky-cell-resizable';
+const STICKY_CELL_FIRST_OF_ROW_CLASS = 'sticky-cell-first-of-row';
+const STICKY_CELL_LAST_OF_COLUMN_CLASS = 'sticky-cell-last-of-column';
+const STICKY_CELL_BORDER_STYLE = `1px solid ${borderColor.default}`;
+
+class Table extends React.PureComponent {
+    constructor() {
+        super();
+        this.state = {
+            adjacentStickyColumnsTotalWidth: 0,
+            stickyTableContainerWidth: 0,
+        };
+        this.applyStickyClassesToCells = this.applyStickyClassesToCells.bind(this);
+        this.delayedSetScrollBarPosition = null;
+        this.forceTableUpdate = this.forceTableUpdate.bind(this);
+        this.parseChildren = this.parseChildren.bind(this);
+        this.setResizableCellsWordWrapping = this.setResizableCellsWordWrapping.bind(this);
+        this.setStickyCellsStylesOnScroll = this.setStickyCellsStylesOnScroll.bind(this);
+    }
+
+    componentDidMount() {
+        this.setStickyColumnPositions();
+        this.applyStickyClassesToCells();
+        this.delayedSetStickyTableContainerWidth = debounce(() => {
+            // eslint-disable-next-line max-len
+            this.setState({
+                stickyTableContainerWidth: this.tableStickyContainer.clientWidth,
+            });
+            this.setStickyColumnPositions();
+        }, DEBOUNCE_WAIT_TIME);
+    }
+
+    componentDidUpdate(prevProps) {
+        const {
+            children: prevChildren,
+            stickyColumnCount: prevStickyColumnCount,
+        } = prevProps;
+        const {
+            children: nextChildren,
+            stickyColumnCount: nextStickyColumnCount,
+        } = this.props;
+        if (prevChildren !== nextChildren || prevStickyColumnCount !== nextStickyColumnCount) {
+            this.setStickyColumnPositions();
+            this.applyStickyClassesToCells();
+        }
+    }
+
+    setStickyColumnPositions() {
+        const {
+            stickyColumnCount,
+            definition,
+            basic,
+            fullWidth,
+            singleLine,
+            celled,
+            fixed,
+        } = this.props;
+
+        if (stickyColumnCount > 0 && this.tableRef) {
+            const stickyCells = this.tableRef.querySelectorAll(`.table--cell:nth-child(-n+${stickyColumnCount})`);
+            let cellWidths = 0;
+            let cellCount = 0;
+            /**
+             * Reducing each left sticky column position will help to not have a separation
+             * between them, so the content behind won't be visible when scrolling
+             */
+            let leftOverFloatSpace = 1;
+            let auxLeftOverFloatSpace = 0;
+            if (basic) {
```

Comment:
NAB: There is a comment somewhere around that @groberts314 made about tending to like whitespace, having blank empty lines between things like constants. This makes reading just a tid-bit easier, but totally not a blocker.

Reply:
I'll apply a blank space before each `if` statement trying not to get the `Block must not be padded by blank lines.` and/or `Block must not be padded by blank lines.` ESLint issues.